hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/has/hederaaccountnumalias/HederaAccountNumAliasCall.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.exec.systemcontracts.has.hederaaccountnumalias;
+
+import static com.hedera.hapi.node.base.ResponseCodeEnum.INVALID_SOLIDITY_ADDRESS;
+import static com.hedera.hapi.node.base.ResponseCodeEnum.SUCCESS;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.FullResult.successResult;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.common.Call.PricedResult.gasOnly;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.has.hederaaccountnumalias.HederaAccountNumAliasTranslator.HEDERA_ACCOUNT_NUM_ALIAS;
+import static com.hedera.node.app.service.contract.impl.exec.systemcontracts.hts.ReturnTypes.ZERO_ADDRESS;
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.accountNumberForEvmReference;
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.asEvmAddress;
+import static com.hedera.node.app.service.contract.impl.utils.ConversionUtils.asHeadlongAddress;
+import static java.util.Objects.requireNonNull;
+
+import com.esaulpaugh.headlong.abi.Address;
+import com.hedera.hapi.node.base.ResponseCodeEnum;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.FullResult;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.common.AbstractCall;
+import com.hedera.node.app.service.contract.impl.exec.systemcontracts.has.HasCallAttempt;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public class HederaAccountNumAliasCall extends AbstractCall {
+    private final Address address;
+
+    public HederaAccountNumAliasCall(@NonNull final HasCallAttempt attempt, @NonNull final Address address) {
+        super(attempt.systemContractGasCalculator(), attempt.enhancement(), true);
+        this.address = requireNonNull(address);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public @NonNull PricedResult execute() {
+
+        final long accountNum = accountNumberForEvmReference(address, nativeOperations());
+        if (accountNum < 0) {
+            // Invalid: not an alias for any account
+            return gasOnly(fullResultsFor(INVALID_SOLIDITY_ADDRESS, ZERO_ADDRESS), INVALID_SOLIDITY_ADDRESS, true);
+        }
+        final var account = requireNonNull(enhancement.nativeOperations().getAccount(accountNum));
```

Comment:
For this and the next line would it not be better to check for the null or non existent account and return a response code?  As it is, this will throw exceptions and revert the call.

Reply:
Hmm. I copied this from elsewhere (e.g., see `HbarAllowanceCall.getAllowance` @L72).  Should that be fixed too?

Reply:
(answered: yes) See #15314

Reply:
Now I remember what happened.  Turns out the call to `accountNumberForEvmReference` method called at L52 above this _does_ the null check, and a null account there causes the `INVALID_SOLIDITY_ADDRESS` at L55.  The `requireNonNull` was is to shut IntelliJ up, because it doesn't know that.

But in fact since it confused you, and confused me too when I went back to look at it, I'll put an explicit (though redundant) check here so the question never again arises.